lib.js

```diff
@@ -41,17 +41,29 @@ module.exports = (options) => {
         });
     };
 
-    const uploadChallenge = (key, value, repo, domain) => {
+    const uploadChallenge = (key, value, repo) => {
         // Need to bluebird-ify to use .asCallback()
-        return Promise.resolve(gitlabRequest.post({
-            url: `/projects/${repo.id}/repository/files`,
-            body: {
-                file_path: `public/.well-known/acme-challenge/${key}`,
-                commit_message: 'Automated Let\'s Encrypt renewal',
-                branch_name: 'master',
-                content: value
-            }
-        })).return([`http://${domain}/.well-known/acme-challenge/${key}`, value]);
+        if ( options.jekyll == "true" ) {
+        return Promise.resolve(gitlabRequest.put({
+              url: `/projects/${repo.id}/repository/files`,
+              body: {
+                  file_path: `${options.sitepath}/letsencrypt.html`,
```

Comment:
Instead of providing a Jekyll-specific `sitepath` option, it might be more useful to provide a `path` option that lets you specify the full path where the LE challenge file will be uploaded.